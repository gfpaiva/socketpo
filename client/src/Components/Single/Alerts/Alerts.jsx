import React, { Fragment } from 'react';

import { Link } from 'react-router-dom';

import Loading from '../../Loading/Loading';

const Alerts = ( { type } ) => (
	<div className="page page--full page--centered page--bg-gradient page--single">
		<div>
			{type === 'Loading' && (
				<Fragment>
					<Loading />
					<p>Loading, please wait</p>
				</Fragment>
			)}

			{type === 'Not Found' && (
				<Fragment>
					<p>
						Match not found <span role='img' aria-label='think'>ðŸ¤”</span><br />
						<Link className='link link--underline' to='/create'>Create a new match</Link>
					</p>
				</Fragment>
			)}

			{type === 'Full' && (
				<Fragment>
					<p>
						This match is full and in progress <br />
						<Link className='link link--underline' to='/create'>Create a new match</Link>
					</p>
				</Fragment>
			)}
		</div>
	</div>
);

export default Alerts;
