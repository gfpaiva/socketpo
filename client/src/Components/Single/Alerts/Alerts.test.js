import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Alerts from './Alerts';

describe('<Alerts />', () => {

	it('should mount "loading" properly', () => {
		const wrapper = mount(
			<Router>
				<Alerts
					type="Loading"
				/>
			</Router>
		);

		expect(wrapper.find('Alerts')).toMatchSnapshot();
	});

	it('should mount "not found" properly', () => {
		const wrapper = mount(
			<Router>
				<Alerts
					type="Not Found"
				/>
			</Router>
		);

		expect(wrapper.find('Alerts')).toMatchSnapshot();
	});

	it('should mount "full" properly', () => {
		const wrapper = mount(
			<Router>
				<Alerts
					type="Full"
				/>
			</Router>
		);

		expect(wrapper.find('Alerts')).toMatchSnapshot();
	});
});
