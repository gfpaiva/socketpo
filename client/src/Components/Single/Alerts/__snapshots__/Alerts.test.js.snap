// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alerts /> should mount "full" properly 1`] = `
<Alerts
  type="Full"
>
  <div
    className="page page--full page--centered page--bg-gradient page--single"
  >
    <div>
      <p>
        This match is full and in progress 
        <br />
        <Link
          className="link link--underline"
          replace={false}
          to="/create"
        >
          <a
            className="link link--underline"
            href="/create"
            onClick={[Function]}
          >
            Create a new match
          </a>
        </Link>
      </p>
    </div>
  </div>
</Alerts>
`;

exports[`<Alerts /> should mount "loading" properly 1`] = `
<Alerts
  type="Loading"
>
  <div
    className="page page--full page--centered page--bg-gradient page--single"
  >
    <div>
      <Loading>
        <Loader
          active={true}
          type="ball-pulse"
        >
          <div
            className="loader loader-undefined loader-active"
          >
            <div
              className="loader-inner ball-pulse"
            >
              <div
                key="0"
                style={Object {}}
              />
              <div
                key="1"
                style={Object {}}
              />
              <div
                key="2"
                style={Object {}}
              />
            </div>
          </div>
        </Loader>
      </Loading>
      <p>
        Loading, please wait
      </p>
    </div>
  </div>
</Alerts>
`;

exports[`<Alerts /> should mount "not found" properly 1`] = `
<Alerts
  type="Not Found"
>
  <div
    className="page page--full page--centered page--bg-gradient page--single"
  >
    <div>
      <p>
        Match not found 
        <span
          aria-label="think"
          role="img"
        >
          ðŸ¤”
        </span>
        <br />
        <Link
          className="link link--underline"
          replace={false}
          to="/create"
        >
          <a
            className="link link--underline"
            href="/create"
            onClick={[Function]}
          >
            Create a new match
          </a>
        </Link>
      </p>
    </div>
  </div>
</Alerts>
`;
